Domain/ProductManagement/Inventory.cs

```diff
@@ -0,0 +1,117 @@
+﻿namespace LenaInventoryManagementSystem;
+
+public class Inventory
+{
+    private static List<Product> _products = new();
+    private static InventoryUI _inventoryUI=new();
+    private const int _maxItemsInStock = 150;
+
+    internal static void MainMenu()
+    {
+        _inventoryUI.DisplayMainMenu();
+        var userSelection=_inventoryUI.UserSelection();
+
+        switch (userSelection)
+        {
+             case "1":
+                _inventoryUI.AddingANewProductUI();
+                break;
+            case "2":
+                _inventoryUI.DisplayAllProducts();
+                break;
+            case "3":
+                _inventoryUI.UpdateProductUI();
+                break;
+            case "4":
+                _inventoryUI.DeletingAProductUI();
+                break;
+            case "5":
+                _inventoryUI.ViewAProductUI();
+                break;
+            default:
+                _inventoryUI.PrintMessage("Invalid selection. Please try again.");
+                break;
+        }
+    }
+
+    internal static void AddANewProduct(string name, double price, int quantity)
+    {
+        //Checking if the product exists.
+        Product? existingProduct = _products.FirstOrDefault(p => p.ProductName == name);
+
+        //Creatng a new product if the product doesn't exists.
+        if (quantity <= _maxItemsInStock && quantity > 0)
+        {
+            var product = new Product(name, price, quantity);
+            _products.Add(product);
+        }
+    }
+
+    internal static List<Product> ViewAllProducts()
+    {
+        return _products;
+    }
+
+    internal static void UpdateProduct(string name, int choice)
+    {
+        Product? existingProduct = _products.FirstOrDefault(p => p.ProductName == name);
+
+        if (existingProduct != null)
+        {
+            switch (choice)
+            {
+                case 1:
```

Comment:
If i read this method alone, I won't understand what these numbers refer to, lets pass an enum value instead.